frontend/src/features/documents/documentDetail/DocumentDetailView.tsx

```diff
@@ -16,17 +16,13 @@ interface IDocumentDetailsViewProps {
  * Component that provides functionality to see document information. Can be embedded as a widget.
  */
 const DocumentDetailView: React.FunctionComponent<IDocumentDetailsViewProps> = props => {
-  let documentTypeLabel = '';
-  let documentFileName = '';
-
-  let mayanDocumentId = -1;
-  let mayanFileId = -1;
+  const documentTypeLabel = props.document.pimsDocument?.documentType?.documentType;
+  const documentFileName = props.document.pimsDocument?.fileName;
+  const mayanDocumentId = props.document.pimsDocument?.mayanDocumentId || -1;
 
+  let mayanFileId = undefined;
   if (props.document.mayanMetadata !== undefined && props.document.mayanMetadata?.length > 0) {
     const document = props.document.mayanMetadata[0].document;
-    documentTypeLabel = document.document_type.label;
-    documentFileName = document.label;
-    mayanDocumentId = document.id;
     mayanFileId = document.file_latest.id;
```

Comment:
There's still a chance for NPE if `document.mayanMetadata[0] == null` 

perhaps consider:
```tsx
const document = props.document.mayanMetadata[0]?.document;
```